VirtoCommerce.MarketingModule.Test/DynamicPromotionTests.cs

```diff
@@ -14,8 +14,20 @@ namespace VirtoCommerce.MarketingModule.Test
 {
     public class DynamicPromotionTests
     {
-        [Fact]
-        public void FindValidCouponTest()
+        public static IEnumerable<object[]> Data =>
+            new List<object[]>
+            {
+                new object[] {9, 10, null, 0},
+                new object[] {10, 10, null, 0},
+                new object[] {11, 10, null, 1},
+                new object[] {11, 1, DateTime.Now.AddDays(-1), 0},
+                new object[] {11, 1, DateTime.Now, 1},
+                new object[] {11, 1, DateTime.Now.AddDays(1), 1},
+            };
+
+        [Theory]
+        [MemberData(nameof(Data))]
```

Comment:
I think better to have two test methods to not mix-up the logic and make atomic test:
FindValidCoupon_UsesNumber
FindValidCoupon_ExpirationData
MS best practices:
https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices